arctic/arctic.py

```diff
@@ -408,7 +408,8 @@ def __init__(self, arctic, library):
         database_name, library = self._parse_db_lib(library)
         self.library = library
         self.database_name = database_name
-        self._auth(self.arctic._conn[self.database_name])
+        self._auth(self._curr_conn[self.database_name])  # authenticate
+        self.get_top_level_collection()  # allow early caching of the library in Arctic (assures _reset is called)
```

Comment:
Not sure this can make a difference as the various library `__init__` methods we have call `get_top_level_collection()` implicitly.  So I suspect something else is going on...

Reply:
certainly no effect here, just updated for sake of simplicity to avoid an extra unnecessary call, since arctic._conn is a property wrapper